include/boost/python/detail/value_is_xxx.hpp

```diff
@@ -9,24 +9,25 @@
 # include <boost/mpl/bool.hpp>
 # include <boost/preprocessor/enum_params.hpp>
 
-
-#  include <boost/type_traits/remove_reference.hpp>
-#  include <boost/type_traits/remove_cv.hpp>
+# include <boost/python/detail/type_traits.hpp>
 #  include <boost/python/detail/is_xxx.hpp>
 
-#  define BOOST_PYTHON_VALUE_IS_XXX_DEF(name, qualified_name, nargs)    \
-template <class X_>                                                     \
-struct value_is_##name                                                  \
-{                                                                       \
-    BOOST_PYTHON_IS_XXX_DEF(name,qualified_name,nargs)                  \
-    BOOST_STATIC_CONSTANT(bool, value = is_##name<                      \
-                               typename remove_cv<                      \
-                                  typename remove_reference<X_>::type   \
-                               >::type                                  \
-                           >::value);                                   \
-    typedef mpl::bool_<value> type;                                    \
-                                                                        \
+namespace boost { namespace python { namespace detail {
+
+#  define BOOST_PYTHON_VALUE_IS_XXX_DEF(name, qualified_name, nargs)                           \
+template <class X_>                                                                            \
+struct value_is_##name                                                                         \
+{                                                                                              \
+    BOOST_PYTHON_IS_XXX_DEF(name,qualified_name,nargs)                                         \
+    BOOST_STATIC_CONSTANT(bool, value = is_##name<                                             \
+                               typename boost::python::detail::remove_cv<                      \
+                                  typename boost::python::detail::remove_reference<X_>::type   \
```

Comment:
Last minor nit-pick: now that this entire template is within the boost::python::detail namespace, it appears we don't need the qualification of `remove_cv` and `remove_reference` in the last two lines any longer.